lib/workers/global/autodiscover.ts

```diff
@@ -51,10 +51,15 @@ export async function autodiscoverRepositories(
 
   if (autodiscoverFilter) {
     logger.debug({ autodiscoverFilter }, 'Applying autodiscoverFilter');
-    discovered = applyFilters(
-      discovered,
-      is.string(autodiscoverFilter) ? [autodiscoverFilter] : autodiscoverFilter,
-    );
+    const addFilters = is.string(autodiscoverFilter)
+      ? [autodiscoverFilter]
+      : autodiscoverFilter;
+    const removeFilters = autodiscoverExclusions
+      ? is.string(autodiscoverExclusions)
+        ? [autodiscoverExclusions]
+        : autodiscoverExclusions
+      : [];
+    discovered = applyFilters(discovered, addFilters, removeFilters);
```

Comment:
```suggestion
    const addFilters = prepareFilter(autodiscoverFilter)
    const removeFilters = prepareFilter(autodiscoverExclusions)
    discovered = applyFilters(discovered, addFilters, removeFilters);
```

extract this into a separate function for readability. 

```typescript
function prepareFilter(filter: string | string[] | undefined): string[] {
  if ((is.array(filter))) {
    return filter
  }
  if (is.string(filter)) {
    return [filter]
  }
  return []
}
```